### ISSUE TITLE
whitelist for antivirus vendors

### ISSUE THREAD

**Issue by az0** [ActivityWatch/activitywatch#181]
I saw in past issues  (e.g., #140 ) that people reported that the software is flagged by antivirus software, and that this is reasonable estimate based on the heuristic of a keylogger. 

The latest Windows zip is [flagged by 9 scanners](https://www.virustotal.com/#/file/320a3ed29ee87fa9de5b497f396f597ce51b74d91a5bbb9673d68ad5b7cbdb8b/detection). 

Would you please consider contacting the vendors to whitelist it? There is a contact list on [techsupportalert](https://www.techsupportalert.com/content/how-report-malware-or-false-positives-multiple-antivirus-vendors.htm) and another list is available if you email VirusTotal.

**Comment by ErikBjare** [comment]
I don't think this needs to be done by the maintainers ourselves, so please feel free to do so on your own! :slightly_smiling_face:

**Comment by 1000i100** [comment]
Positive vote to send : 
[activitywatch-v0.7.1-windows-x86_64.zip](https://www.virustotal.com/#/file/320a3ed29ee87fa9de5b497f396f597ce51b74d91a5bbb9673d68ad5b7cbdb8b/community)
[aw-server.exe](https://www.virustotal.com/#/file/1b10655acbe1c57494a1fd140b63665f1eacd09be038e9f8901b6dd213c2c720/community)
[aw-qt.exe](https://www.virustotal.com/#/file/3aa17c8920198784ac3b63f9bfae38995cf94749f68659f030b4500b1b426e8e/community)
[aw-watcher-window.exe](https://www.virustotal.com/#/file/cccde227d2b4d6a39b4481e6f1185b1eec71ea232f422e377d0d2800584afb1a/community)
[aw-watcher-afk.exe](https://www.virustotal.com/#/file/6cf7aa271617984beef508d485d4752f8b6272f8e45f990f956ffd2c034f98b4/community)

**Comment by pcuci** [comment]
![image](https://user-images.githubusercontent.com/2169849/69083482-2ed9c680-0a10-11ea-927f-243c61a52653.png)

Well this is annoying :-)

**Comment by johan-bjareholt** [comment]
@pcuci Please submit a false positive to Microsoft about that, we can't do anything else about it than that as Windows doesn't provide any safe APIs for us to use.

**Comment by pcuci** [comment]
For what it's worth, the admin team at my work managed to add an exception, then asked me to execute the following steps to clear the cached detection and obtain the latest malware definitions.

1. Open command prompt as administrator and change directory to
c:\Program Files\Windows Defender 
2. Run "MpCmdRun.exe -removedefinitions -dynamicsignatures" 
3. Run "MpCmdRun.exe -SignatureUpdate"

It appears that IT help-desks inside organizations have the ability to include exceptions. I don't know if these new malware definitions later go upstream to Microsoft, it may very well be the case, or not.

Hope this encourages others to negotiate with their IT/network/security teams :-)

**Comment by johan-bjareholt** [comment]
> It appears that IT help-desks inside organizations have the ability to include antivirus exceptions.

The only annoying thing is that if you ever update ActivityWatch you will likely have to go through the same procedure again.

**Comment by ErikBjare** [comment]
A more long-term solution might be to code-sign the releases, or simply put it up on the Windows store.

**Comment by wasinix** [comment]
I am not able to run release 0.8.4 on my office pc
McAffee Endpoint Security is declaring ActivityWatch as Ransomware and blocking processes and partially deleting files (aw-watcher-afk.exe)

Found several entries in eventlog from McAffee, including details, what he thinks is evil, but as the log are in german, I dont know if posting them here makes sense.

**Comment by ErikBjare** [comment]
Apparently AlternativeTo now shows a malware warning for ActivityWatch (reported in #493). Not sure what we can do about that.

However, someone dropped this link on the AlternativeTo page which gives a lot of nice details about *why* it's considered suspicious: https://www.hybrid-analysis.com/sample/beb047cb7583df66301493c613afe0d7bf6c62b5445eb38797b6fcf38d239afe/5e7cd780c49eaf4be46cde62

But alas, it only confirms what we already knew: it's all guesswork.

**Comment by Julian** [comment]
Suspicious indeed.

**Comment by ErikBjare** [comment]
AlternativeTo replied to my email and have removed the warning. Thanks @timharek for reporting!

**Comment by rakleed** [comment]
@ErikBjare @johan-bjareholt I submitted .exe files in various ways (especially aw-qt.exe, aw-server.exe, and others) to McAfee, but got no response. Then Tier 2 support agent sent me [this article](https://kc.mcafee.com/corporate/index?page=content&id=KB85568). You can try to contact them via datasubmission@mcafee.com.

**Comment by rakleed** [comment]
I scanned all the files on VirusTotal and then reported false positives to anti-virus vendors for several months. As a result, most vendors have responded and fixed the issues, but some just haven't answered. The results can be viewed here: [GitHub Gist - activitywatch_virustotal](https://gist.github.com/rakleed/b1f17cc378eacf55abb87025ff96d080) and [backup link](https://web.archive.org/web/20210727091146/https://gist.github.com/rakleed/b1f17cc378eacf55abb87025ff96d080) (although I haven't updated them for the last few weeks).

I tried to contact vendors using information from [this repository](https://github.com/yaronelh/False-Positive-Center) (I also updated some data in it myself).

I also tried to solve the problem through [VirusTotal support](https://www.virustotal.com/gui/contact-us), but they helped at first, and then they began to ignore my requests.

But you can also try to contact them, perhaps due to the large number of complaints, they will still correct false positives.

**Comment by nck974** [comment]
Hello I was using version v0.10 and tried to update to v0.11 and I get the following in firefox:
![image](https://user-images.githubusercontent.com/18548911/129200909-6126f45f-66c3-4344-b969-92656f75b66f.png)

**Comment by tbertels** [comment]
![image](https://user-images.githubusercontent.com/3265870/135503068-599e2d45-c946-46eb-b46b-9599000072d1.png)
https://virusscan.jotti.org/en-US/filescanjob/clmvm45bbb

But still banned in Firefox and Chrome.

Based on https://developers.google.com/search/docs/advanced/security/malware it seems that the developer has to follow this procedure (Security Issues report
): https://support.google.com/webmasters/answer/9044101

**Comment by MaxJW** [comment]
Similar issue found again for v0.12.2, was about to create a ticket but saw this!
Is there a procedure for resolving this now?

![image](https://user-images.githubusercontent.com/15789924/228799846-0d9c0526-6502-4e80-a6bb-bd61e1ffc191.png)

**Comment by rakleed** [comment]
@MaxJW https://www.microsoft.com/en-us/wdsi/filesubmission

**Comment by ErikBjare** [comment]
@rakleed Thanks for linking, I just submitted `v0.12.2`

**Comment by rdggithub** [comment]
Avast flags v0.12.3b15

**Comment by J05HM0N5TER** [comment]
Windows defender, activity watch version [v0.13.0](https://github.com/ActivityWatch/activitywatch/releases/tag/v0.13.0). Was detected as `Trojan:Script/Wacatac.B!ml`, specificly the `aw-qt.exe` file. It was triggered during the install/update process, and was removed.

**Comment by john30** [comment]
> Windows defender, activity watch version [v0.13.0](https://github.com/ActivityWatch/activitywatch/releases/tag/v0.13.0). Was detected as `Trojan:Script/Wacatac.B!ml`, specificly the `aw-qt.exe` file. It was triggered during the install/update process, and was removed.

same here. is there something suspicious in the https://github.com/ActivityWatch/aw-qt ? didn't find something obious in the commits since v0.12.2 though, only updates to PyQt6 6.5.3 and pyinstaller 6.6 might be candidates I'd say

**Comment by ErikBjare** [comment]
`trojan:Script/Wacatac.B!ml` is a machine-learning guided rule (the `!ml` suffix) and commonly a false positive: https://superuser.com/a/1830913/247123

**Comment by ErikBjare** [comment]
I've submitted the v0.13.1 release to Microsoft as a false positive for Windows Defender.

**Comment by lyc8503** [comment]
Usually code-signing can suppress these false positives from Machine learning rules. 

However #632 is not resolved till now, and the releases I got from scoop are still unsigned and flagged as malware (10/79 on VirusTotal)

@ErikBjare Are there still plans to purchase certificates? If so, I'm guessing all we're missing is a CI configuration for code signing. Maybe I can issue a PR to complete the CI on GitHub Actions.

**Comment by ErikBjare** [comment]
@lyc8503 I will happily buy the cert if you or anyone else set up the necessary CI. 

But please do set it up self-signed before I pay for the cert, a couple people have previously offered/attempted but churned on the task :)

**Comment by lyc8503** [comment]
> But please do set it up self-signed before I pay for the cert, a couple people have previously offered/attempted but churned on the task :)

You're right, there's no guarantee I can get this done, but I'll give it a try next week when I have time!

**Comment by lyc8503** [comment]
@ErikBjare Just made an attempt to sign the executables in PR #1092. Hopefully someone can test it, it looks like there are some other issues with the build process on the master branch right now.

**Comment by Neurone** [comment]
[Avast and others](https://www.virustotal.com/gui/file/666c4d7a7bd9346d9d714bedd17b7dee5ed98e66fb277de3c026c088de6d6627) still consider the latest Windows installer `activitywatch-v0.13.2-windows-x86_64-setup.exe` infected.
